Mathlib/Topology/Category/FinTopCat.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.Topology.Category.TopCat.Basic
+
+/-!
+# Category of finite topological spaces
+
+Definition of the category of finite topological spaces with the canonical
+forgetful functors.
+
+-/
+
+
+universe u
+
+open CategoryTheory
+
+/-- A bundled finite topological space. -/
+structure FinTopCat where
+  /-- carrier of a finite topological space. -/
+  toTop : TopCat.{u}
+  [fintype : Fintype toTop]
+
+namespace FinTopCat
+
+instance : Inhabited FinTopCat :=
+  ⟨{ toTop := { α := PEmpty } }⟩
+
+instance : CoeSort FinTopCat (Type u) :=
+  ⟨fun X => X.toTop⟩
+
+attribute [instance] fintype
+
+instance : Category FinTopCat :=
+  InducedCategory.category toTop
+
+instance : ConcreteCategory FinTopCat :=
+  InducedCategory.concreteCategory _
+
+instance (X : FinTopCat) : TopologicalSpace ((forget FinTopCat).obj X) :=
+  inferInstanceAs <| TopologicalSpace X
+
+instance (X : FinTopCat) : Fintype ((forget FinTopCat).obj X) :=
+  X.fintype
```

Comment:
Should the spelling be `(forget _).obj X` or `X` (from the `CoeSort` instance)?

Reply:
If this does not break things, I would favour the `CoeSort` versoin.

Reply:
The `CoeSort` version already works:
```
example (X : FinTopCat) : Fintype X := inferInstance
```
but
```
example (X : FinTopCat) : Fintype ((forget FinTopCat).obj X) := inferInstance
```
does not without the instance above and I think I'll need this one.

Reply:
It's sad that we need that. I really hope we can sort out the coercions of concrete categories one day.

Reply:
Should I do anything about this in this case or can we merge this PR then?